service/resources/submission.py

```diff
@@ -15,18 +17,44 @@ class Submission(Base):
     id = sa.Column('id', sa.Integer, primary_key=True)
     data = sa.Column('data', sa.Text, nullable=False)
     date_created = sa.Column('date_created', sa.DateTime(timezone=True), server_default=func.now())
+    external_ids = relationship("ExternalId")
+
+    dispatch_count = {}
+
+class ExternalId(Base):
+    __tablename__ = 'external_id'
+    id = sa.Column('id', sa.Integer, primary_key=True)
+    submission_id = sa.Column('submission_id', sa.Integer, sa.ForeignKey('submission.id'))
+    external_id = sa.Column('external_id', sa.Text, nullable=False)
+    external_system = sa.Column('external_system', sa.VARCHAR(length=255), nullable=False)
+    date_created = sa.Column('date_created', sa.DateTime(timezone=True), server_default=func.now())
 
 @falcon.before(validate_access)
 class SubmissionResource:
     
     def on_post(self, req, resp):
         # log submission to database
-        submission = Submission(data=json.dumps(req.params))
-        self.session.add(submission)
-        self.session.commit()
+        submission = create_submission(self.session, json.dumps(req.params))
+        # schedule dispatch to external systems
+        jobs_scheduled = jobs.schedule(submission_obj=submission, systems_dict=jobs.external_systems)
```

Comment:
What if this job scheduled failed? Should we have a way to make sure jobs are scheduled properly for submissions?

Reply:
You mean like if something were to happen to redis and the job isn't added to the queue?

Reply:
Yeah

Reply:
api returns 500 error if db or redis are down.

https://github.com/SFDigitalServices/adu-dispatcher-microservice-py/pull/4/commits/08e5dc4599ed8a93b78120e3dd22f440049e8d56